testnet/bitcoin-neon-controller/src/main.rs

```diff
@@ -0,0 +1,296 @@
+    let stream = TcpStream::connect(config.neon.bitcoind_rpc_host.clone()).await?;
+    let body = serde_json::to_vec(&req).unwrap();
+    let mut resp = client::connect(stream.clone(), build_request(&config, body)).await?;
+
+    let (res, buffer) = async_std::task::block_on(async move {
+        let mut buffer = Vec::new();
+        let mut body = resp.take_body();
+        let res = body.read_to_end(&mut buffer).await;
+        (res, buffer)
+    });
+
+    // Guard: can't be read 
+    if res.is_err() {
+        panic!("Chain height could not be determined")
+    }
+    // let mut deserializer = Deserializer::from_slice(&buffer);
+
+    let mut deserializer = Deserializer::from_slice(&buffer);
+
+    // Guard: can't be parsed
+    let rpc_resp: RPCResult = match RPCResult::deserialize(&mut deserializer) {
+        Ok(rpc_req) => rpc_req,
+        _ => panic!("Chain height could not be determined")
+    };
+
+    match (rpc_resp.result, rpc_resp.error) {
+        (Some(_), None) => return Ok(false),
+        (None, Some(error)) => {
+            if let Some(keys) = error.as_object() {
+                if let Some(message) = keys.get("message") {
+                    if let Some(message) = message.as_str() {
+                        if message == "Block height out of range" {
+                            return Ok(true)
+                        }
+                    }
+                }
+            }
+        }
+        (_, _) => {}
+    }
+
+    panic!("Chain height could not be determined")
+}
+
+async fn generate_blocks(blocks_count: u64, address: String, config: &ConfigFile) {
+    let rpc_addr = config.neon.bitcoind_rpc_host.clone();
+
+    let rpc_req = RPCRequest::generate_next_block_req(blocks_count, address);
+
+    let stream = TcpStream::connect(rpc_addr).await.unwrap();
+    let body = serde_json::to_vec(&rpc_req).unwrap();
+    let req = build_request(&config, body);
+    client::connect(stream.clone(), req).await.unwrap();
+}
+
+fn build_request(config: &ConfigFile, body: Vec<u8>) -> Request {
+    let url = Url::parse(&format!("http://{}/", config.neon.bitcoind_rpc_host)).unwrap();
+    let mut req = Request::new(Method::Post, url);
+    req.append_header("Authorization", config.neon.authorization_token()).unwrap();
+    req.append_header("Content-Type", "application/json").unwrap();
+    req.append_header("Content-Length", format!("{}", body.len())).unwrap();
+    req.append_header("Host", format!("{}", config.neon.bitcoind_rpc_host)).unwrap();
+    req.set_body(body);
+    req
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+/// JSONRPC Request
+pub struct RPCRequest {
+    /// The name of the RPC call
+    pub method: String,
+    /// Parameters to the RPC call
+    pub params: Vec<serde_json::Value>,
+    /// Identifier for this Request, which should appear in the response
+    pub id: serde_json::Value,
+    /// jsonrpc field, MUST be "2.0"
+    pub jsonrpc: Option<String>,
```

Comment:
If this `MUST be "2.0"`, should this just be a `String`?

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1518/commits/8ff29538c69f3ba2ce121336c6ec087be3d812bc